packages/material-ui/src/Typography/Typography.test.js

```diff
@@ -70,38 +97,144 @@ describe('<Typography />', () => {
+      const wrapper = shallow(
+        <Typography color={color} useNextVariants>
+          Hello
+        </Typography>,
+      );
       assert.ok(classes[className] !== undefined);
       assert.strictEqual(wrapper.hasClass(classes[className]), true, `should be ${color} text`);
     });
   });
 
   describe('prop: color', () => {
     it('should inherit the color', () => {
-      const wrapper = shallow(<Typography color="inherit">Hello</Typography>);
+      const wrapper = shallow(
+        <Typography color="inherit" useNextVariants>
+          Hello
+        </Typography>,
+      );
       assert.strictEqual(wrapper.hasClass(classes.colorInherit), true);
     });
   });
 
   describe('headline', () => {
     it('should render a span by default', () => {
-      const wrapper = shallow(<Typography variant="button">Hello</Typography>);
+      const wrapper = shallow(
+        <Typography variant="button" useNextVariants>
+          Hello
+        </Typography>,
+      );
       assert.strictEqual(wrapper.name(), 'span');
     });
 
     it('should render a p with a paragraph', () => {
-      const wrapper = shallow(<Typography paragraph>Hello</Typography>);
+      const wrapper = shallow(
+        <Typography paragraph useNextVariants>
+          Hello
+        </Typography>,
+      );
       assert.strictEqual(wrapper.name(), 'p');
     });
 
     it('should render the mapped headline', () => {
-      const wrapper = shallow(<Typography variant="title">Hello</Typography>);
-      assert.strictEqual(wrapper.name(), 'h2');
+      const wrapper = shallow(<Typography variant="headline6">Hello</Typography>);
+      assert.strictEqual(wrapper.name(), 'h6');
     });
 
     it('should render a h1', () => {
-      const wrapper = shallow(<Typography component="h1">Hello</Typography>);
+      const wrapper = shallow(
+        <Typography component="h1" useNextVariants>
+          Hello
+        </Typography>,
+      );
       assert.strictEqual(wrapper.name(), 'h1');
     });
   });
+
+  describe('v2 migration', () => {
+    const mount = createMount();
+    let warning;
+
+    beforeEach(() => {
+      warning = mock(console).expects('error');
+    });
+
+    afterEach(() => {
+      warning.restore();
+    });
+
+    /**
+     * tests if a warning is issued from the `warning` module when mounting {component}
+     */
+    const testMount = (component, expectDeprecation) => {
+      const expectedWarning = expectDeprecation ? 'Deprecation Warning: Material-UI:' : undefined;
+      warning.resetHistory();
+
+      try {
+        const theme = withEnabledWarnings();
+        const wrapper = mount(<MuiThemeProvider theme={theme}>{component}</MuiThemeProvider>);
+        wrapper.unmount();
+
+        if (expectedWarning) {
+          assert.fail('got no error', `expected a warning to match '${expectedWarning}'`);
+        }
+      } catch (e) {
+        assert.isTrue(expectDeprecation, 'expected no deprecation but got a warning anyway');
```

Comment:
There is only one occurrence of `assert.isTrue` in the code base (one that you added). What do you think of using `strictEqual(x, true)`? At least it's minimizing the entropy.

Reply:
I learned that you should always use the appropriate matcher methods for readability. I don't see why I should not use the matchers that are available.

Reply:
I think that it's a tradeoff between having meaningful error messages vs controlling your assertion. `assert.strictEqual` leaves no room for doubt, I think that it's very important, it's a brainless `===`. For instance, Is `isTrue` using strict or nonstrict equal?
The exceptions are `.deepEqual(`, `.throw(` and `.include(` because using strict equal there requires too much boilerplate or a very poor debug message when it fails.

Reply:
Named matchers (with `is` or `be`) use strict equality in chai, jest and jasmine. Those are the most popular assertion libraries in js. I don't agree that there is room for doubt if we are talking about code in a unit test.

I'm not arguing about this particular line of code but I think there is a reason assertion libraries even include noop chain methods to further improve readability in unit tests. 

I think my main issue is chai's assert library since I'm used to jest which is much better to read and the discussion whether to use `toBe(true)` or `toBeTrue()` would be fruitless (funny enough jest doesn't have `toBeTrue` but `toBe[Null|Undefined]`).

I'll change it for consistency.